manager/allocator/network.go

```diff
@@ -1052,6 +1052,9 @@ func (a *Allocator) allocateNode(ctx context.Context, node *api.Node, existingAd
 		// for every attachment, go through every network. if the attachment
 		// belongs to one of the networks, then go to the next attachment. if
 		// no network matches, then the the attachment should be removed.
+		if attach.Network == nil {
+			continue
+		}
```

Comment:
Should arguably be placed before the comment on lines 1052-1054, since that comment is about what follows this.